bin/download_images.go

```diff
@@ -11,40 +11,93 @@ import (
   "path/filepath"
   "regexp"
   "strings"
+  "unicode"
 )
 
 type Opts struct {
-  Directory string `short:"d" long:"directory" description:"The absolute path of the dir to scan" required:"true"`
-  ImageDir  string `short:"i" long:"image-dir" description:"The absolute path of where to put images" required:"true"`
+  Directory string `short:"d" long:"directory" description:"The absolute path of the dir to scan, try '-d $PWD'" required:"true"`
+  ImageDir  string `short:"i" long:"image-dir" description:"The absolute path of where to put images, try '-i $PWD/images/'" required:"true"`
+  CheckOnly bool   `long:"check" description:"Exits non zero on images not hosted by Armory or stored in the repo" required:"false"`
 }
 
+
 func main() {
   var opts Opts
-
   _, err := flags.Parse(&opts)
   if err != nil {
     os.Exit(1)
   }
 
-  fmt.Println("Looking for images to download")
+  switch {
+  case opts.CheckOnly:
+    imgUrls := checkForImgsNotHostedByArmory(opts)
+    if len(imgUrls) > 0 {
+      fmt.Printf("Found %d images that are external to this this repo or Armory.\n\n", len(imgUrls))
+      for _, s := range imgUrls {
+        fmt.Println(s)
+      }
+      fmt.Println(`
+You can download them by doing:
+  docker run -v $(pwd):/src --workdir /src golang:1.12.0 go run -mod=vendor bin/download_images.go -d /src -i /src/images/`)
+      os.Exit(1)
+    }
+  default:
+    fmt.Println("Looking for images to download")
+    _ = filepath.Walk(opts.Directory, rewriteMarkdownWithSavedImages(opts))
+  }
+
+  return
+}
+
+const markdownImageUrlRegex = `!\[.*]\((https?://.*)\)`
+
+func checkForImgsNotHostedByArmory(opts Opts) []string {
+  var imageUrls []string
 
   _ = filepath.Walk(opts.Directory, func(filePath string, info os.FileInfo, err error) error {
-    if !strings.Contains(filePath, ".md") && !strings.Contains(filePath, ".html") {
+    if !shouldExploreFilePath(filePath) {
       return nil
     }
 
     fileContents, _ := ioutil.ReadFile(filePath)
 
-    markdownImageUrlMatch := regexp.MustCompile(`!\[.*\]\((https?://.*)\)`)
+    markdownImageUrlMatch := regexp.MustCompile(markdownImageUrlRegex)
 
     if markdownImageUrlMatch.Match(fileContents) {
       lines := strings.Split(string(fileContents), "\n")
 
+      for _, line := range lines {
+        if markdownImageUrlMatch.MatchString(line) {
+          imgs := markdownImageUrlMatch.FindStringSubmatch(line)
+          imageUrls = append(imageUrls, imgs[1:]...)
```

Comment:
how come you don't append all images?

Reply:
The first match is the entire string ``![...](....)` and then the 2nd is the http...